src/Util/BigInteger.php

```diff
@@ -203,4 +231,44 @@ public function compare(BigInteger $y)
     {
         return gmp_cmp($this->value, $y->value);
     }
+
+    /**
+     * Check whether two BigIntegers are the same.
+     *
+     * @param BigInteger $x
+     *
+     * @return bool
+     */
+    public function equals(BigInteger $x)
+    {
+        return 0 === gmp_cmp($this->value, $x->value);
```

Comment:
You are right but in this case the function is used to compute primes and not to compare values during a signature verification.

If think an attack could be mitigated if:

* RSA keys without primes are not used (in this case this code is not reached)
* RSA keys without primes are refused. Primes have to be calculated first.